app/services/public_transportation/import_service.rb

```diff
@@ -0,0 +1,192 @@
+      route_id = parsed_line["route_id"].to_i
+      next unless route_ids_of_trips.include?(route_id.to_s)
+
+      RedisAdapter.set_gtfs_route(route_id, parsed_line, @data_provider_id)
+      route_counter += 1
+    end
+    p "Found #{route_counter} Routes in routes.txt"
+  end
+
+  def parse_trips(zip_file)
+    p "Parsing trips.txt"
+    trip_counter = 0
+    trip_ids_of_stop_times = RedisAdapter.get_gtfs_trip_ids(@data_provider_id).map(&:to_i)
+    parse_file_of_zip(zip_file, "trips.txt") do |parsed_line|
+      trip_id = parsed_line["trip_id"].to_i
+      # next unless trip_ids_of_stop_times.include?(trip_id)
+
+      RedisAdapter.set_gtfs_trip(trip_id, parsed_line, @data_provider_id)
+      RedisAdapter.set_gtfs_route_ids(parsed_line["route_id"], @data_provider_id)
+      trip_counter += 1
+    end
+    p "Found #{trip_counter} trips in trips.txt"
+  end
+
+  def parse_stop_times(zip_file)
+    p "Parsing stop_times.txt"
+    stop_counter = 0
+    parse_file_of_zip(zip_file, "stop_times.txt") do |parsed_line|
+      stop_id = parsed_line["stop_id"].to_i
+      next unless @pois.keys.include?(stop_id)
+
+      RedisAdapter.set_gtfs_trip_ids(parsed_line["trip_id"].to_i, @data_provider_id)
+      RedisAdapter.set_gtfs_stop_times(stop_id, parsed_line, @data_provider_id)
+      stop_counter += 1
+    end
+
+    p "Found #{stop_counter} stops in stop_times.txt"
+  end
+
+  # select all stops from stops.txt
+  def parse_stops(zip_file)
+    p "Parsing stops.txt"
+    parse_file_of_zip(zip_file, "stops.txt") do |parsed_line|
+      poi_stop_id = parsed_line["stop_id"].to_i
+      next if @whitelist.present? && @whitelist.any? && !@whitelist.include?(poi_stop_id)
+
+      # create a new point of interest or update it
+      point_of_interest = PointOfInterest.where(external_id: poi_stop_id.to_s).first_or_initialize
+      point_of_interest.name = parsed_line["stop_name"].force_encoding("UTF-8")
+      point_of_interest.data_provider_id = @data_provider_id
+      point_of_interest.payload = parsed_line.to_h
+      point_of_interest.save
+      @pois[poi_stop_id] = point_of_interest.id
+
+      p "Updating POI id: #{point_of_interest.id}, poi_stop_id: #{poi_stop_id}"
+    end
+  end
+
+  # method to pars a csv file from a zip file with a given file name
+  # and a given block to call with the parsed line
+  def parse_file_of_zip(zip_file, file_name)
+    data_file = zip_file.glob(file_name).first
+    raise "File #{file_name} not found in zip file" if data_file.nil?
+
+    header_data = nil
+    data_file.get_input_stream.read.force_encoding("UTF-8").each_line.with_index do |line, index|
+      # set csv header and skip in all following lines
+      (header_data = CSV.parse(line).first; next) if index.zero?
+
+      # parse line with header_data_line as csv header
+      parsed_line = CSV.parse(line, headers: header_data).first
+
+      # Call the given block with the parsed line
+      yield(parsed_line) if block_given?
+    end
+  end
```

Comment:
def parse_file_of_zip ist eine wiederverwendbare Methode, die aus dem ZipFile eine einzelne, konkrete Datei entpackt und jede Zeile in der Datei durchgeht.
 
 Beim verwenden oberhalb wird ein Block übergeben in der Form:

 ```
  parse_file_of_zip(zip_file, "stops.txt") do |parsed_line|
    p parsed_line
  end 
 ```
 
 der Block wird dann der Stelle `yield(parsed_line)` eingefügt/ausgeführt